tests/hazmat/primitives/test_dh.py

```diff
@@ -473,6 +474,39 @@ def test_public_key_copy(self):
 
         assert key1 == key2
 
+    def test_old_public_key_exchange(self, backend):
+        # This is a public key generated on cryptograpy version 41.0.7
+        key_data = textwrap.dedent(
```

Comment:
Generally we use keys from the `vectors/` directory in our tests -- there should be an existing one with a matching format.